src/routes/edit-invoice.tsx

```diff
@@ -57,45 +69,31 @@ export default function ViewInvoice() {
     setStatus('loading');
 
     try {
-      await InvoicesService.updateInvoice(id, data);
+      await updateInvoiceMutation.mutateAsync({ id, data });
       setStatus('success');
     } catch (error) {
       console.log('error');
       setStatus('error');
     }
   };
 
-  const handleSave = () => {
-    console.log('Saving invoice...');
-    handleSubmit(onSubmit); // WHY NO WORK ?!?!
-  };
-
-  const handleEdit = () => {
-    if (isEditMode) {
-      console.log('Canceling edit...');
-      navigate(`/invoice/${id}`);
-    } else {
-      navigate(`/invoice/${id}/edit`);
-      console.log('Editing invoice...');
-    }
-    setIsEditMode((prevState) => !prevState);
-  };
-
   return (
     <div>
       <Box sx={{ m: 4 }}>
         <h1 style={visuallyHidden}>{t('INVOICE.TITLE')}</h1>
       </Box>
 
-      <form>
+      <form onSubmit={handleSubmit(onSubmit)}>
```

Comment:
I think you should disable the native validation, otherwise it may become strange UX, as you will be adding zod validation :D